File: frontend/demo/component/passwordfield/password-field-constraints.ts
import 'Frontend/demo/init'; // hidden-source-line

import { html, LitElement } from 'lit';
import { customElement } from 'lit/decorators.js';
import '@vaadin/password-field';
import { applyTheme } from 'Frontend/generated/theme';

@customElement('password-field-pattern')
export class Example extends LitElement {
  protected override createRenderRoot() {
    const root = super.createRenderRoot();
    // Apply custom theme (only supported if your app uses one)
    applyTheme(root);
    return root;
  }

  protected override render() {
    return html`
      <!-- tag::snippet[] -->
      <vaadin-password-field
        allowed-char-pattern="[A-Za-z0-9]"
        required
        min-length="6"
        max-length="12"
        label="Password"
        helper-text="6 to 12 characters. Only letters A-Z and numbers supported."
      ></vaadin-password-field>
      <!-- end::snippet[] -->
    `;
  }
}

File: src/main/java/com/vaadin/demo/component/passwordfield/PasswordFieldConstraints.java
package com.vaadin.demo.component.passwordfield;

import com.vaadin.flow.component.orderedlayout.HorizontalLayout;
import com.vaadin.flow.component.textfield.PasswordField;
import com.vaadin.flow.router.Route;
import com.vaadin.demo.DemoExporter; // hidden-source-line

@Route("password-field-constraints")
public class PasswordFieldConstraints extends HorizontalLayout {

    public PasswordFieldConstraints() {
        setPadding(false);

        // tag::snippet[]
        PasswordField field = new PasswordField("Password");
        field.setRequired(true);
        field.setAllowedCharPattern("[A-Za-z0-9]");
        field.setMinLength(6);
        field.setMaxLength(12);
        // end::snippet[]
        field.setHelperText("6 to 12 characters. Only letters A-Z and numbers supported.");

        add(field);
    }

    public static class Exporter extends DemoExporter<PasswordFieldConstraints> { // hidden-source-line
    } // hidden-source-line
}
